pkg/controller/clusterautoscaler/clusterautoscaler_controller.go

```diff
@@ -57,6 +57,13 @@ type Config struct {
 	Verbosity int
 }
 
+// deploymentUpdatedInterface is used to enable unit testing and does not
+// implement all methods of Reconciler.
+type deploymentUpdatedInterface interface {
```

Comment:
Not totally sure about the value of the interface here. Couldn't  we create a fake runtime client and test against multiple dep autoscaler specs and also use the value ["release.openshift.io/version"] coming from there so we use it to validate cases? cc @bison

Reply:
Agreed. Don't want to be a pain here, but I think it's important we keep the strategy aligned with our other operators and the other OpenShift operators. That's generally testing inputs and outputs via the fake k8s client rather than overriding individual methods for different test cases.